plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js

```diff
@@ -224,51 +238,82 @@ export default class ChatEmojiPicker extends Component {
-        .closest(".chat-emoji-picker__section")
-        .querySelectorAll(".emoji"),
+    const sectionsEmojis = (section) => [...section.querySelectorAll(".emoji")];
+    const focusSectionsLastEmoji = (section) => {
+      const emojis = sectionsEmojis(section);
+      return emojis[emojis.length - 1].focus();
+    };
+    const focusSectionsFirstEmoji = (section) => {
+      sectionsEmojis(section)[0].focus();
+    };
+    const currentSection = event.target.closest(".chat-emoji-picker__section");
+    const focusFilter = document.querySelector(".dc-filter-input")?.focus();
+    const allEmojis = () => [
+      ...document.querySelectorAll(
+        ".chat-emoji-picker__scrollable-content .emoji"
+      ),
     ];
 
     if (event.key === "ArrowRight") {
       event.preventDefault();
+      const nextEmoji = event.target.nextElementSibling;
 
-      if (event.target === sectionEmojis[sectionEmojis.length - 1]) {
-        sectionEmojis[0].focus();
+      if (nextEmoji) {
+        nextEmoji.focus();
       } else {
-        event.target.nextElementSibling?.focus();
+        const nextSection = currentSection.nextElementSibling;
+        if (nextSection) {
+          focusSectionsFirstEmoji(nextSection);
+        }
       }
     }
 
     if (event.key === "ArrowLeft") {
       event.preventDefault();
+      const prevEmoji = event.target.previousElementSibling;
 
-      if (event.target === sectionEmojis[0]) {
-        sectionEmojis[sectionEmojis.length - 1].focus();
+      if (prevEmoji) {
+        prevEmoji.focus();
       } else {
-        event.target.previousElementSibling?.focus();
+        const prevSection = currentSection.previousElementSibling;
+        if (prevSection) {
+          focusSectionsLastEmoji(prevSection);
+        } else {
+          focusFilter();
+        }
       }
     }
 
     if (event.key === "ArrowDown") {
       event.preventDefault();
       event.stopPropagation();
 
-      sectionEmojis
+      const nextEmoji = allEmojis()
         .filter((c) => c.offsetTop > event.target.offsetTop)
-        .find((c) => c.offsetLeft === event.target.offsetLeft)
-        ?.focus();
+        .findBy("offsetLeft", event.target.offsetLeft);
+
+      if (nextEmoji) {
+        nextEmoji.focus();
+      } else {
+        // for perf reason all emojis might not be loaded at this point
+        // but the first one will always be
+        const nextSection = currentSection.nextElementSibling;
+        if (nextSection) {
+          focusSectionsFirstEmoji(nextSection);
+        }
+      }
     }
 
     if (event.key === "ArrowUp") {
       event.preventDefault();
       event.stopPropagation();
 
-      sectionEmojis
+      const prevEmoji = allEmojis()
         .reverse()
         .filter((c) => c.offsetTop < event.target.offsetTop)
-        .find((c) => c.offsetLeft === event.target.offsetLeft)
-        ?.focus();
+        .findBy("offsetLeft", event.target.offsetLeft);
+
+      prevEmoji?.focus() || focusFilter();
```

Comment:
Same issue here, I believe it should be:

```suggestion
      prevEmoji?.focus() || focusFilter;
```

Reply:
weird... not it's the opposite, focusFilter should be made a function, will change :+1: